src/Metadata/Property/Factory/ConstructorPropertyMetadataFactory.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Metadata\Property\Factory;
+
+use ApiPlatform\Core\Exception\PropertyNotFoundException;
+use ApiPlatform\Core\Metadata\Property\PropertyMetadata;
+
+/**
+ * Set properties available in the constructor as writable.
+ *
+ * @author Maxime Veber <maxime.veber@nekland.fr>
+ */
+class ConstructorPropertyMetadataFactory implements PropertyMetadataFactoryInterface
+{
+    /**
+     * @var PropertyMetadataFactoryInterface
+     */
+    private $decorated;
+
+    public function __construct(PropertyMetadataFactoryInterface $decorated = null)
+    {
+        $this->decorated = $decorated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(string $resourceClass, string $property, array $options = []): PropertyMetadata
+    {
+        if (null === $this->decorated) {
+            $propertyMetadata = new PropertyMetadata();
+        } else {
+            try {
+                $propertyMetadata = $this->decorated->create($resourceClass, $property, $options);
+            } catch (PropertyNotFoundException $propertyNotFoundException) {
+                $propertyMetadata = new PropertyMetadata();
+            }
+        }
+
+        // Constructor arguments are obviously accessible only on post operation, put will result in an error.
+        if (!isset($options['collection_operation_name']) || 'post' !== $options['collection_operation_name']) {
```

Comment:
What if we have custom collection operations that don't have the `post` name?

Reply:
I don't understand the use case, can you be more specific? It doesn't make sense to allow constructor in another request type than post.

Reply:
He meant this:
```
collectionOperations={
    "foo" => {"method"="POST"}
}
```

Reply:
I guess the best way to fix this is to add the method to the context, what do you think about?

Reply:
Huum, `ResourceMetadata` should do the job. For example:
```php
$this->resourceMetadataFactory->create($resourceClass)->getCollectionOperationAttribute($options['collection_operation_name'], 'method')
```

Reply:
Fixed.